src/Features/CSharp/Portable/Copilot/CSharpCopilotNotImplementedMethodFixProvider.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Copilot;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using static Microsoft.CodeAnalysis.CodeActions.CodeAction;
+
+namespace Microsoft.CodeAnalysis.CSharp.Copilot;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.ImplementNotImplementedException), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed partial class CSharpCopilotNotImplementedMethodFixProvider() : SyntaxEditorBasedCodeFixProvider
+{
+    public override ImmutableArray<string> FixableDiagnosticIds { get; }
+        = [IDEDiagnosticIds.CopilotImplementNotImplementedExceptionDiagnosticId];
+
+    public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var document = context.Document;
+        var cancellationToken = context.CancellationToken;
+
+        if (document.GetRequiredLanguageService<ICopilotOptionsService>() is not { } copilotOptionsService ||
+            await copilotOptionsService.IsGenerateMethodImplementationOptionEnabledAsync().ConfigureAwait(false) is false)
+        {
+            return;
+        }
+
+        if (document.GetLanguageService<ICopilotCodeAnalysisService>() is not { } copilotService ||
+            await copilotService.IsAvailableAsync(cancellationToken).ConfigureAwait(false) is false)
+        {
+            return;
+        }
+
+        // Find the throw statement or throw expression node
+        var throwNode = FindThrowNode(context.Diagnostics[0], cancellationToken);
+        if (throwNode is null)
+            return;
+
+        // Preliminary analysis before registering fix
+        var methodDeclaration = throwNode.Ancestors().OfType<MethodDeclarationSyntax>().FirstOrDefault();
+        if (methodDeclaration != null)
+        {
+            if (methodDeclaration.ExpressionBody != null)
+            {
+                // Arrow expression body not supported yet
+                return;
```

Comment:
also, why not support this?  it should be very cheap/easy to do so.  this is also a mainline case as many users will get this by default when doing sometehing like implement-interface if they prefer expression-bodied members.

Reply:
added support